client/components/webpack-build-monitor/index.jsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-console */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createStore } from 'redux';
+import classNames from 'classnames';
+import { find, includes, startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Spinner from 'components/spinner';
+
+const CONNECTED = 'CONNECTED';
+const DISCONNECTED = 'DISCONNECTED';
+const NEEDS_RELOAD = 'NEEDS_RELOAD';
+const BUILDING_CSS = 'BUILDING_CSS';
+const BUILDING_JS = 'BUILDING_JS';
+const BUILDING_BOTH = 'BUILDING_BOTH';
+const ERROR = 'ERROR';
+const IDLE = 'IDLE';
+
+const doneBuilding = typeDone => state => {
+	if ( state === ERROR || state === NEEDS_RELOAD ) {
+		return state;
+	} else if ( state === BUILDING_BOTH ) {
+		return typeDone === BUILDING_JS ? BUILDING_CSS : BUILDING_JS;
+	}
+	return IDLE;
+};
+
+const buildJS = state => ( state === BUILDING_CSS ? BUILDING_BOTH : BUILDING_JS );
+const buildCSS = state => ( state === BUILDING_JS ? BUILDING_BOTH : BUILDING_CSS );
```

Comment:
fixed the nit.  

and in terms of NEEDS_RELOAD, I ended up doing something pretty clean in the reducer.  Once NEEDS_RELOAD is hit, then it'll stay on it forever